frontend/packages/dev-console/src/components/topology2/components/nodes/WorkloadNode.tsx

```diff
@@ -43,12 +44,21 @@ const WorkloadNode: React.FC<WorkloadNodeProps> = ({
   const { width, height } = element.getBounds();
   const workloadData = element.getData().data;
   const size = Math.min(width, height);
-  const { donutStatus } = workloadData;
+  const { donutStatus, editUrl, cheEnabled } = workloadData;
   const { radius, decoratorRadius } = calculateRadius(size);
-  const repoIcon = routeDecoratorIcon(workloadData.editUrl, decoratorRadius);
   const cx = width / 2;
   const cy = height / 2;
-
+  const repoIcon =
+    cheEnabled && editUrl ? (
+      <CheIcon
+        x={cx - radius * 1.07}
+        y={cx - radius * 1.02}
+        width={radius * 0.39}
+        height={radius * 0.31}
```

Comment:
Why all these magic numbers?
The PF icons have a height and width of `1em` so we control them using the font-size and set it to the radius.
I would expect the width and height here to also be the radius and the che svg itself be adjusted to fit appropriately similar to the PF icons.
The fact that the width and height aren't the same here means the icon is distorted no?

Reply:
@christianvogt The icon is not distorted , i have updated the code to have width and height as 1em and adjusting the cheIcon now using the fontsize similar to the PF Icons.